### PR TITLE
Fixes #30638 - replace react-numeric-input with rc-number-input

### PR DESCRIPTION
**Pull request by yifatmakias**
The rc-input-number is a much simple library and has fewer bugs therefore it should replace the react-numeric-input in the NumericInput react component.

### CODE REVIEW

**Comment by yifatmakias** [comment]
This PR is a blocker PR to:
https://github.com/theforeman/foreman/pull/7794
https://github.com/theforeman/foreman/pull/7865

**Comment by yifatmakias** [comment]
[test foreman]

File: webpack/assets/javascripts/react_app/components/common/forms/NumericInput.scss
Comment (sharvit) [inline webpack/assets/javascripts/react_app/components/common/forms/NumericInput.scss]: It would be nice to use a different class-name if possible, something that `.foreman-numeric-input`, `.foreman-numeric-input-input-wrap`, `.foreman-numeric-input-handler-up-inner` and so on...

Looks like the `rc-input-number` supports the field `prefixCls`, can we use that?
https://github.com/react-component/input-number#props

File: package.json
Comment (sharvit) [inline package.json]: Need to be added to [`@theforeman/vendor`](https://github.com/theforeman/foreman-js/tree/master/packages/vendor).

I just wrote this doc yesterday and as I promised last week, I will do a deep-dive about the topic soon:
https://github.com/theforeman/foreman-js/pull/197

You are welcome to ping me for help and comment on my PR if something isn't clear enough.

**Review by sharvit — CHANGES_REQUESTED** [review]
Thanks @yifatmakias, works well 👍 

Doesn't it going to break to following css files?
https://github.com/theforeman/foreman/blob/2cd4d88297a36d9a43a34a64badb3b7aa18e7d31/webpack/assets/javascripts/react_app/components/CPUCoresInput/cpuCoresInput.scss
https://github.com/theforeman/foreman/blob/ae96cb70130968bdcc8b7a9a9af7edc7c6e0d9a8/webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/disk.scss

File: package.json
Comment (yifatmakias) [reply package.json:26]: I think I added it to @theforeman/vendor. let me know if there is still something missing.

**Comment by yifatmakias** [comment]
[test foreman]

**Comment by yifatmakias** [comment]
This PR is blocked by another PR:
https://github.com/theforeman/foreman-js/pull/198

**Comment by yifatmakias** [comment]
[test katello]

File: package.json
Comment (sharvit) [reply package.json]: theforeman/foreman-js#198 is merged so you can remove this line and update all `@theforeman/*` packages to use v4.14.0, see:
https://github.com/theforeman/foreman-js/pull/198#issuecomment-674844385

**Review by sharvit — CHANGES_REQUESTED** [review]
Thanks @yifatmakias

**Comment by sharvit** [comment]
packaging-pr: https://github.com/theforeman/foreman-packaging/pull/5687

File: webpack/assets/javascripts/react_app/components/common/forms/NumericInput.js
Comment (sharvit) [inline webpack/assets/javascripts/react_app/components/common/forms/NumericInput.js]: I could find two dives with the `foreman-numeric-input` class-name on the dom.
I guess only one of them should receive the css properties defined in the scss file?

File: webpack/assets/javascripts/react_app/components/common/forms/NumericInput.scss
Comment (sharvit) [inline webpack/assets/javascripts/react_app/components/common/forms/NumericInput.scss]: Are you sure the `!important` is needed?

File: webpack/assets/javascripts/react_app/components/common/forms/NumericInput.scss
Comment (sharvit) [inline webpack/assets/javascripts/react_app/components/common/forms/NumericInput.scss:3]: You add `@import "~@theforeman/vendor/scss/variables";` to the beginning of the file to have access to pf3/pf4 scss variables that you can use instead hard-coded values.

IMO it makes sense to reuse them for colors, shadows, sizes, and spaces.

Those are all the variables and the mixins available for use:
- https://theforeman-vendor-docs-v4-14-2.surge.sh/scss/variables.scss
- https://theforeman-vendor-docs-v4-14-2.surge.sh/scss/mixins.scss

It might be easier to search using the patternfly docs:
- PF3 - https://www.patternfly.org/v3/styles/color-palette/
- PF4 - https://www.patternfly.org/v4/documentation/overview/global-css-variables

Comment (yifatmakias) [reply webpack/assets/javascripts/react_app/components/common/forms/NumericInput.scss:3]: I added this import and change the colors and some of the sizes. Let me know if I missed some changes.

File: webpack/assets/javascripts/react_app/components/common/forms/NumericInput.scss
Comment (sharvit) [inline webpack/assets/javascripts/react_app/components/common/forms/NumericInput.scss:12]: We could drop the input styles if we could set the input className to `form-control`.
Unfortunately, the `rc-input-number` doesn't support custom class names on the input field.

Sent a PR to `rc-input-number` so they can have this feature:
https://github.com/react-component/input-number/pull/262

Can you please add a comment in the code with a reminder to drop the input styles once merged?

File: webpack/assets/javascripts/react_app/components/common/forms/NumericInput.scss
Comment (sharvit) [inline webpack/assets/javascripts/react_app/components/common/forms/NumericInput.scss:31]: When I set the disabled prop to true, the hover effect still exists and it looks odd.

File: webpack/assets/javascripts/react_app/components/common/forms/NumericInput.scss
Comment (sharvit) [inline webpack/assets/javascripts/react_app/components/common/forms/NumericInput.scss]: I don't see the glowing effect that we usually have when focused.
try:
```scss
@import "~@theforeman/vendor/scss/mixins";

&:focus {
  @include form-control-outline;
}
```

**Review by sharvit — CHANGES_REQUESTED** [review]
Thanks @yifatmakias

**Review by sharvit — DISMISSED** [review]
Thanks @yifatmakias, looks good and works as expected 👍

**Comment by sharvit** [comment]
Can we get ack from packaging please?

File: .stylelintrc
Comment (ekohl) [inline .stylelintrc]: Personally I'm a fan of trailing commas because they always result in smaller diffs when you add/remove lines. Not sure if they're actually valid in JS though.

**Review by ekohl — DISMISSED** [review]
Packaging ACK, didn't really look at the actual JS code.

File: .stylelintrc
Comment (yifatmakias) [reply .stylelintrc]: I added the commas as they were before

File: .stylelintrc
Comment (ekohl) [inline .stylelintrc]: I'd be consistent and also add it here. You may also want to consider sorting the list alphabetically. Those are just some practices I started to use because it's easier to deal with when things get longer and you're searching for something.

**Review by ekohl — DISMISSED** [review]
Still a packaging ack.

File: .stylelintrc
Comment (yifatmakias) [reply .stylelintrc]: @ekohl Added another comma and also sorted alphabetically. Thanks for the review! :)

**Review by ekohl — APPROVED** [review]
LGTM. The tests are still running now, but otherwise I think it can be merged.

**Comment by ekohl** [comment]
Oh, they finished. @sharvit I'm not sure you gave an ACK or not so leaving the final honors of merging to you.

**Review by sharvit — APPROVED** [review]
Thanks @yifatmakias, looks good and works as expected 👍 

Thanks @ekohl, merging.

**Pull request merged**
